code/game/world.dm

```diff
@@ -164,10 +166,30 @@ GLOBAL_VAR(restart_counter)
 	log_runtime(GLOB.revdata.get_log_message())
 
 /world/Topic(T, addr, master, key)
-	TGS_TOPIC	//redirect to server tools if necessary
-
+	var/static/list/bannedsourceaddrs = list()
+	var/static/list/lasttimeaddr = list()
 	var/static/list/topic_handlers = TopicHandlers()
 
+	//LEAVE THIS COOLDOWN HANDLING IN PLACE, OR SO HELP ME I WILL MAKE YOU SUFFER
+	if (addr in bannedsourceaddrs)
+		return

```

Comment:
Use an associated list reference instead of `in`.

`in` is O(n), `blah[key]` is O(log n)

Reply:
for some reason I thought that gives a runtime about the key if it doesn't exist